src/API/src/db/shared/reference.resolver.ts

```diff
@@ -0,0 +1,18 @@
+import { Args, Query, Resolver } from '@nestjs/graphql';
+import { ReferenceService } from './reference.service';
+import { Reference } from './entities/reference.entity';
+
+@Resolver(() => Reference)
+export class ReferenceResolver {
+  constructor(private referenceService: ReferenceService) {}
+
+  @Query(() => Reference)
+  async referenceData(@Args('id', { type: () => String }) id: string) {
+    return this.referenceService.findOneById(id);
+  }
+
+  @Query(() => [Reference])
+  async allReferenceData() {
+    return this.referenceService.findAll();
```

Comment:
The acceptance criteria for this story (https://github.com/icipe-official/vectoratlas-software-code/issues/67) mention adding infinite scroll - is there a plan to add this in later?